### ISSUE TITLE
Causing ip leak after reboot node cause multus cache file damaged

### ISSUE THREAD

**Issue by hangzhouwanjun** [k8snetworkplumbingwg/multus-cni#1087]
**What happend**:
Using lastest release multus 4.01，calico as the underlying delegate cni，macvlan as second cni，host-local as ipam. 
NetworkAttachmentDefinition such as：
```
config: '{
  "cniVersion": "0.3.0",
  "name": "net-111-master1",
  "master": "eth2",
  "type": "macvlan",
  "ipam": {
          "type": "host-local",
          "ranges": [[{"subnet":"1.1.0.0/16","gateway":"1.1.0.1"}]]
          }
 }' 
```
Node reboot cause multus cache file damaged, pod status error because of ip failed to allocate for range 0
pod describe：
`Failed to create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "3e9da5542cbd12a2b23820fdbd7ba1043f4eda6e30b0657b2a0926b8aeaed3b4" network for pod "vcfc-1-b667d84cf-lv5sn": networkPlugin cni failed to set up pod "vcfc-1-b667d84cf-lv5sn_appcluster" network: [appcluster/vcfc-1-b667d84cf-lv5sn/:net-111-master1]: error adding container to network "net-111-master1": failed to allocate for range 0: requested IP address 1.1.0.4 is not available in range set 1.1.0.1-1.1.255.254`

Multus code has some error if multus cache file not JSON format, multus return nil not call calico、macvlan、host-local cmdDel cause ip leak，pod status error。
```
// CmdDel ...
func CmdDel(args *skel.CmdArgs, exec invoke.Exec, kubeClient *k8s.ClientInfo) error {
...
        netconfBytes, path, err := consumeScratchNetConf(args.ContainerID, in.CNIDir)
	useCacheConf := false
	if err == nil {
		in.Delegates = []*types.DelegateNetConf{}
		if err := json.Unmarshal(netconfBytes, &in.Delegates); err != nil {
			logging.Errorf("Multus: failed to load netconf: %v", err)
		} else {
			useCacheConf = true
			// check plugins field and enable ConfListPlugin if there is
			for _, v := range in.Delegates {
				if len(v.ConfList.Plugins) != 0 {
					v.ConfListPlugin = true
				}
			}
			// First delegate is always the master plugin
			in.Delegates[0].MasterPlugin = true
		}
	}

	if !useCacheConf {
		// Fetch delegates again if cache is not exist and pod info can be read
		if os.IsNotExist(err) && pod != nil {
			if in.ClusterNetwork != "" {
				_, err = k8s.GetDefaultNetworks(pod, in, kubeClient, nil)
				if err != nil {
					return cmdErr(k8sArgs, "failed to get clusterNetwork/defaultNetworks: %v", err)
				}
				// First delegate is always the master plugin
				in.Delegates[0].MasterPlugin = true
			}

			// Get pod annotation and so on
			_, _, err := k8s.TryLoadPodDelegates(pod, in, kubeClient, nil)
			if err != nil {
				if len(in.Delegates) == 0 {
					// No delegate available so send error
					return cmdErr(k8sArgs, "failed to get delegates: %v", err)
				}
				// Get clusterNetwork before, so continue to delete
				logging.Errorf("Multus: failed to get delegates: %v, but continue to delete clusterNetwork", err)
			}
		} else {
			// The options to continue with a delete have been exhausted (cachefile + API query didn't work)
			// We cannot exit with an error as this may cause a sandbox to never get deleted.
			logging.Errorf("Multus: failed to get the cached delegates file: %v, cannot properly delete", err)
			return nil
		}
	} 
```
Multus cache file damaged not JSON format，useCacheConf will false and return nil，but calico、macvlan、host-local not cmdDel，causing ip leak，pod status error

**What you expected to happen**:
Reboot node even if multus cache file damaged, no ip leak even if multus cache file damaged

**How to reproduce it (as minimally and precisely as possible)**:
Yes, if the /var/lib/cni/multus/bc50ca19f75360f83f91470a39a20ada04170bb4620b3c9ccbcdd3de367b6a95 is empty or not JSON format. 

**Anything else we need to know?**:

**Environment**:

- Multus version 4.0.1
  image path and image ID (from 'docker images')
- Kubernetes version (use `kubectl version`): 1.21.0
- Primary CNI for Kubernetes cluster: calico
- OS (e.g. from /etc/os-release):
- File of '/etc/cni/net.d/'
- File of '/etc/cni/multus/net.d'
- NetworkAttachment info (use `kubectl get net-attach-def -o yaml`)
- Target pod yaml info (with annotation, use `kubectl get pod <podname> -o yaml`)
- Other log outputs (if you use multus logging)

**Comment by oliver11rz** [comment]
Excuse me, how was this problem solved later? I encountered the same problem. Restarting the machine causes IP leakage, and it prompts that the IP is occupied.

**Comment by lucheng0127** [comment]
Same problem I got
I try to use host-local as ipam, then I try to assign an ip_1 to pod. Same error occur.
But when I try to assign ip_2 to pod, pod launch well.

Here is my NetworkAttachmentDefinition spec
```
spec:
  config: '{"version":"0.3.0","type":"bridge","bridge":"br-net","ipam":{"type":"host-local","subnet":"10.24.0.0/23","rangeStart":"10.24.0.0","rangeEnd":"10.24.1.255","gateway":"10.24.1.254","routes":[{"dst":"0.0.0.0/0"}]},"mtu":1500}'
```
And this my pod.yaml
```
apiVersion: v1
kind: Pod
metadata:
  name: p100
  namespace: kube-system
  annotations:
    k8s.v1.cni.cncf.io/networks: '[{"name":"external-network-0","ips":["10.24.1.10"]}]'
spec:
  containers:
    - name: p100
      image: harbor.litcompute.com/library/alpine:3.20
      command: ["sh"]
      tty: true
```
And its log
```
  Warning  FailedCreatePodSandBox  3s               kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "4f0764571d80e7bf2b8c37d7a34f77b43935eff02bbbfb77f99e395d6742f7c7" network for pod "p100": networkPlugin cni failed to set up pod "p100_kube-system" network: [kube-system/p100/:external-network-0]: error adding container to network "external-network-0": failed to allocate for range 0: requested IP address 10.24.1.10 is not available in range set 10.24.0.0-10.24.1.255
```
When I change the ip of pod, it launch well
```
root@stack-uat-master:~# kubectl get po -n kube-system p100 -w
NAME   READY   STATUS    RESTARTS   AGE
p100   1/1     Running   0          5s
^Croot@stack-uat-master:~# kubectl exec -it -n kube-system p100 ip a
kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl exec [POD] -- [COMMAND] instead.
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: tunl0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN qlen 1000
    link/ipip 0.0.0.0 brd 0.0.0.0
3: net1@if183717: <BROADCAST,MULTICAST,UP,LOWER_UP,M-DOWN> mtu 1500 qdisc noqueue state UP qlen 1000
    link/ether c2:3f:41:cd:a8:e6 brd ff:ff:ff:ff:ff:ff
    inet 10.24.0.22/23 brd 10.24.1.255 scope global net1
       valid_lft forever preferred_lft forever
183715: eth0@if183716: <BROADCAST,MULTICAST,UP,LOWER_UP,M-DOWN> mtu 1400 qdisc noqueue state UP
    link/ether 00:00:00:f3:e7:1b brd ff:ff:ff:ff:ff:ff
    inet 100.64.2.37/10 brd 100.127.255.255 scope global eth0
       valid_lft forever preferred_lft forever
```

**Comment by lucheng0127** [comment]
I got the same problem, but I found if i use host-local,
it seems will check the ip available, if not available it will raise a error
@oliver11rz FYI

Here is the error log
```
  Warning  FailedCreatePodSandBox  7m9s                    kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "fef31563c0e8b4325853c853ad3dea5a83a847b489865d3cce6115a2916d26ff" network for pod "gw-r9n80q0x5752y2-0": networkPlugin cni failed to set up pod "gw-r9n80q0x5752y2-0_kube-system" network: [kube-system/gw-r9n80q0x5752y2-0/:external-network-0]: error adding container to network "external-network-0": failed to allocate for range 0: requested IP address 10.24.1.10 is not available in range set 10.24.0.0-10.24.1.255
```

arping
```
root@tl044:~# arping 10.24.0.10 -c 1
ARPING 10.24.0.10
60 bytes from 52:d5:71:3d:3b:d1 (10.24.0.10): index=0 time=53.247 usec

--- 10.24.0.10 statistics ---
1 packets transmitted, 1 packets received,   0% unanswered (0 extra)
rtt min/avg/max/std-dev = 0.053/0.053/0.053/0.000 ms
root@tl044:~# bridge fdb show |grep 52:d5:71:3d:3b:d1
52:d5:71:3d:3b:d1 dev bond0 master br-net
```

And here is my net-attach-def spec
```
spec:
  config: '{"version":"0.3.0","type":"bridge","bridge":"br-net","ipam":{"type":"host-local","subnet":"10.24.0.0/23","rangeStart":"10.24.0.0","rangeEnd":"10.24.1.255","gateway":"10.24.1.254","routes":[{"dst":"0.0.0.0/0"}]},"mtu":1500}'
```

> Excuse me, how was this problem solved later? I encountered the same problem. Restarting the machine causes IP leakage, and it prompts that the IP is occupied.
